src/server.c

```diff
@@ -193,11 +197,499 @@ struct redisServer server; /* Server global state */
+     "ok-loading ok-stale"},
+
+    {NULL},
+};
+
+struct redisCommand clusterSubcommands[] = {
+    {"addslots",clusterCommand,-3,
+     "admin ok-stale random"},
+
+    {"bumpepoch",clusterCommand,2,
+     "admin ok-stale random"},
+
+    {"count-failure-reports",clusterCommand,3,
+     "admin ok-stale random"},
+
+    {"countkeysinslots",clusterCommand,3,
+     "ok-stale random"},
+
+    {"delslots",clusterCommand,-3,
+     "admin ok-stale random"},
+
+    {"failover",clusterCommand,-2,
+     "admin ok-stale random"},
+
+    {"forget",clusterCommand,3,
+     "admin ok-stale random"},
+
+    {"getkeysinslot",clusterCommand,4,
+     "ok-stale random"},
+
+    {"flushslots",clusterCommand,2,
+     "admin ok-stale random"},
+
+    {"info",clusterCommand,2,
+     "ok-stale random"},
+
+    {"keyslot",clusterCommand,3,
+     "ok-stale random"},
+
+    {"meet",clusterCommand,-4,
+     "admin ok-stale random"},
+
+    {"myid",clusterCommand,2,
+     "ok-stale random"},
+
+    {"nodes",clusterCommand,2,
+     "ok-stale random"},
+
+    {"replicate",clusterCommand,3,
+     "admin ok-stale random"},
+
+    {"reset",clusterCommand,3,
+     "admin ok-stale random"},
+
+    {"set-config-epoch",clusterCommand,3,
+     "admin ok-stale random"},
+
+    {"setslot",clusterCommand,-4,
+     "admin ok-stale random"},
+
+    {"replicas",clusterCommand,3,
+     "admin ok-stale random"},
+
+    {"saveconfig",clusterCommand,2,
+     "admin ok-stale random"},
+
+    {"slots",clusterCommand,2,
+     "ok-stale random"},
+
+    {"help",clusterCommand,2,
+     "ok-loading ok-stale"},
+
+    {NULL},
+};
+
+struct redisCommand sentinelSubcommands[] = {
+    {"ckquorum",sentinelCommand,3,
+     "admin only-sentinel"},
+
+    {"config",sentinelCommand,-3,
+     "admin only-sentinel"},
+
+    {"debug",sentinelCommand,2,
+     "admin only-sentinel"},
+
+    {"get-master-addr-by-name",sentinelCommand,3,
+     "admin only-sentinel"},
+
+    {"failover",sentinelCommand,3,
+     "admin only-sentinel"},
+
+    {"flushconfig",sentinelCommand,2,
+     "admin only-sentinel"},
+
+    {"info-cache",sentinelCommand,3,
+     "admin only-sentinel"},
+
+    {"is-master-down-by-addr",sentinelCommand,6,
+     "admin only-sentinel"},
+
+    {"master",sentinelCommand,3,
+     "admin only-sentinel"},
+
+    {"masters",sentinelCommand,2,
+     "admin only-sentinel"},
+
+    {"monitor",sentinelCommand,6,
+     "admin only-sentinel"},
+
+    {"myid",sentinelCommand,2,
+     "admin only-sentinel"},
+
+    {"pending-scripts",sentinelCommand,2,
+     "admin only-sentinel"},
+
+    {"remove",sentinelCommand,3,
+     "admin only-sentinel"},
+
+    {"replicas",sentinelCommand,3,
+     "admin only-sentinel"},
+
+    {"reset",sentinelCommand,3,
+     "admin only-sentinel"},
+
+    {"sentinels",sentinelCommand,3,
+     "admin only-sentinel"},
+
+    {"set",sentinelCommand,5,
+     "admin only-sentinel"},
+
+    {"simulate-failure",sentinelCommand,3,
+     "admin only-sentinel"},
+
```

Comment:
somehow i think it should be changed like this
```c
    {"set",sentinelCommand,-5,     // -3 -> -5
     "admin only-sentinel"},

    {"simulate-failure",sentinelCommand,-3,     // 3 -> -3
     "admin only-sentinel"},
```

in the past, we can use `sentinel SIMULATE-FAILURE CRASH-AFTER-ELECTION CRASH-AFTER-PROMOTION`, although it is only used for testing, the code supports such use
```
/* SENTINEL SET <mastername> [<option> <value> ...] */
/* SENTINEL SIMULATE-FAILURE <flag> <flag> ... <flag> */
```
and we introduced the sub-command table, so in some sub-commands
so we can not check the number of parameters
in sentinel sub-commands, we can do cleanup like this: https://github.com/enjoy-binbin/redis/commit/0b30a386f7842fd9647fcb7f89fb82532a556b0b
it can remove a `numargserr` label, i think it's worth it...
As for the other sub-commands, i think we can leave them as they are

Reply:
@enjoy-binbin thanks!

i agree with changes in the SENTINEL command table, can you please PR?

regarding the improvements in sentinelCommand : the plan is actually to split it completely into smaller functions, so that each function is the `proc` of a subcommand (same as we did in configCommand). you're welcome to PR that too if you wish.

Reply:
oh i get it, sure i can take care of the SENTINEL command table one, we can hold on the other one

Reply:
just note, we don't wanna split the `sentinelCommand` just yet, it'll mess up the blame log unnecessarily.
we can however do that for `sentinelSetCommand` since it's already split.
but i think it'll look odd that some sub-commands have their own procs, and others don't.